### PR TITLE
Add support for custom claims in `getClaims` function

### PR DESCRIPTION
**Pull request by nicknisi**
This pull request introduces enhancements to JWT handling and testing, with significant updates to the `decodeJwt` function, JWT interfaces, and related utilities. It also refines test cases and improves the overall structure for better maintainability and clarity.

### JWT Handling Enhancements:

* **Added `decodeJwt` utility function**: Introduced a new function to decode JWT tokens, returning both the header and payload. This replaces the previous manual decoding approach and improves error handling. (`src/utils/jwt.ts`, [src/utils/jwt.tsR1-R43](diffhunk://#diff-bc3253061283423c721c833f262624f2ba8b6073381c5122cc6671b500cfc743R1-R43))
* **Updated `getClaims` function**: Refactored the function to use `decodeJwt` instead of manual base64 decoding, simplifying the code and ensuring consistency. (`src/utils/session-data.ts`, [src/utils/session-data.tsL1-R12](diffhunk://#diff-95010fb015f6a0a08b6806392bc2d993462171b43d91c15d4c7ef716710b511bL1-R12))

### Interface Improvements:

* **Defined `JwtHeader` and `JwtPayload` interfaces**: Added detailed type definitions for JWT headers and payloads, improving type safety and documentation. (`src/interfaces/jwt.interface.ts`, [src/interfaces/jwt.interface.tsR1-R64](diffhunk://#diff-80a14d5e1a5cd2f45d3b4328fe2c2cc676665a099d68a154c39d729a84fed34dR1-R64))
* **Exported new JWT interfaces**: Updated exports to include `JwtHeader` and `JwtPayload` for external use. (`src/index.ts`, [[1]](diffhunk://#diff-a2a171449d862fe29692ce031981047d7ab755ae7f84c707aef80701b3ea0c80L3-R9); `src/interfaces/index.ts`, [[2]](diffhunk://#diff-2433a58dd5b382f3ef0faf44d5ab97bf79428a0a1452d72934c9ca3c77a237bcR10)

### Test Coverage Enhancements:

* **Expanded JWT test cases**: Added comprehensive test cases for `decodeJwt`, covering valid tokens, invalid formats, edge cases, and nested structures. (`src/utils/jwt.test.ts`, [src/utils/jwt.test.tsR1-R175](diffhunk://#diff-31dc9de163d99e6f7bdb07ee8ce38416c94089120befeae525eab787b743847bR1-R175))
* **Refined `create-client` tests**: Adjusted test expectations to match the updated JWT handling, ensuring compatibility with the new `decodeJwt` logic. (`src/create-client.test.ts`, [[1]](diffhunk://#diff-0e70951cb32ea9343aa70d26ce568a18783c6bf923d5dcfd2b3a240ac3bc0e42R49-R63) [[2]](diffhunk://#diff-0e70951cb32ea9343aa70d26ce568a18783c6bf923d5dcfd2b3a240ac3bc0e42L674-R680) [[3]](diffhunk://#diff-0e70951cb32ea9343aa70d26ce568a18783c6bf923d5dcfd2b3a240ac3bc0e42L693-R699) [[4]](diffhunk://#diff-0e70951cb32ea9343aa70d26ce568a18783c6bf923d5dcfd2b3a240ac3bc0e42L882-R888)

### CODE REVIEW

File: src/utils/session-data.ts [hunk 1]
```diff
@@ -1,22 +1,15 @@
-import { AuthenticationResponse } from "../interfaces";
+import { AuthenticationResponse, JwtPayload } from "../interfaces";
```
[the remark this answers is not in the document]
Comment (nicknisi) [reply src/utils/session-data.ts:1]: That's not true. It is used.

**Review by cmatheson — APPROVED** [review]
